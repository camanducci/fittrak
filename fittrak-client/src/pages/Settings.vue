<template>
  <v-container>
    <v-layout row>
      <UserSettings :viewer="viewer" />
    </v-layout>
  </v-container>
</template>

<script>
import { queries } from "@/graphql";

import UserSettings from "@/components/settings/UserSettings";

export default {
  name: "SettingsPage",

  data() {
    return {
      viewer: {}
    };
  },

  apollo: {
    viewer: {
      query: queries.viewerQuery
    },

    update(data) {
      return data.viewer;
    }
  },

  components: {
    UserSettings
  }
};
</script>
