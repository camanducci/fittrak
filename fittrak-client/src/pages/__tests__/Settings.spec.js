import { shallowMount } from "@vue/test-utils";
import Settings from "@/pages/Settings";

describe("Settings", () => {
  test("Settings page renders", () => {
    const app = shallowMount(Settings, {
      propsData: {
        viewer: {
          username: "foo",
          email: "foo@baz.com"
        }
      }
    });

    expect(app.html()).toMatchSnapshot();
  });
});
