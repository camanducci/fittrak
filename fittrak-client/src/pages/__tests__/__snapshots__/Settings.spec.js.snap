// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Settings page renders 1`] = `
<v-container-stub tag="div" viewer="[object Object]">
  <v-layout-stub tag="div" row="">
    <usersettings-stub viewer="[object Object]"></usersettings-stub>
  </v-layout-stub>
</v-container-stub>
`;
